src/Agent.Listener/JobDispatcher.cs

```diff
@@ -614,6 +617,35 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, WorkerDisp
                                 var messageType = MessageType.CancelRequest;
                                 if (HostContext.AgentShutdownToken.IsCancellationRequested)
                                 {
+                                    var service = HostContext.GetService<IFeatureFlagProvider>();
+                                    var ffState = await service.GetFeatureFlagAsync(HostContext, "DistributedTask.Agent.FailJobWhenAgentDies", Trace);
```

Comment:
>GetFeatureFlagAsync [](http://example.com/codeflow?start=64&length=19)

This is odd as we're using the knob in other places but here, we're reading the FF using the REST api.  Okay if there is a need to do so but if possible would be good to be consistent

Reply:
Yeah I agree but we can't use Knob to read the Feature Flag in Agent Listener that's why using the REST API to do so. Since REST API is more expensive than reading the value from Knobs, all the other places in Agent Worker read the FF from the Knob.